src/test/java/net/pms/util/FileUtilTest.java

```diff
@@ -120,6 +120,238 @@ public void testGetFileNameWithRewriting() throws Exception {
+        		}
+	        	// tvEpisodeNumber might be a single episode, but might also be
+	        	// a hyphen-separated range, so cannot always parse as int
+	        	String tvEpisodeNumber = extracted_metadata[4];
+	        	String tvEpisodeName = extracted_metadata[5];
+
+        		JsonElement elem;
+
+        		elem = metadata.get("extra");
+        		if( elem != null ) {
+        			for( JsonElement extra: elem.getAsJsonArray() ) {
+        				try {
+        					assertThat( extraInformation.indexOf(extra.getAsString()) > -1 ).isTrue();
+        				}
+        				catch(NullPointerException ex) {
+        					// There is no extraInformation extracted
+        					if(todo) { logger.warn( "testGetFileNameMetadata/extra would fail for TODO test " + original ); }
+        					else { throw(new AssertionError(ex)); }
+        				}
+        				catch(AssertionError err) {
+        					// extraInformation is extracted, but is wrong
+        					if(todo) { logger.warn( "testGetFileNameMetadata/extra would fail for TODO test " + original ); }
+        					else { throw(err); }
+        				}
+        			}
+        		}
+	        	if( "tv-series-episode".equals(metadata.get("type").getAsString()) ) {
+	        		logger.debug( "Doing tv-series-episode " + original );
+	        		// A single episode, might have episode title and date
+	        		elem = metadata.get("series");
+	        		if( elem != null ) {
+	        			try  {
+	        				assertThat(movieOrShowName).isEqualTo(elem.getAsString());
+	        			}
+        				catch(NullPointerException ex) {
+        					// There is no movieOrShowName extracted
+        					if(todo) { logger.warn( "testGetFileNameMetadata/series would fail for TODO test " + original ); }
+        					else { throw(new AssertionError(ex)); }
+        				}
+        				catch(AssertionError err) {
+        					// movieOrShowName is extracted, but is wrong
+        					if(todo) { logger.warn( "testGetFileNameMetadata/series would fail for TODO test " + original ); }
+        					else { throw(err); }
+        				}
+	        		}
+	        		elem = metadata.get("season");
+	        		if( elem != null ) {
+	        			try  {
+		        			assertThat(tvSeason).isEqualTo(elem.getAsInt()); 
+	        			}
+        				catch(NullPointerException ex) {
+        					if(todo) { logger.warn( "testGetFileNameMetadata/season would fail for TODO test " + original ); }
+        					else { throw(new AssertionError(ex)); }
+        				}
+        				catch(AssertionError err) {
+        					if(todo) { logger.warn( "testGetFileNameMetadata/season would fail for TODO test " + original ); }
+        					else { throw(err); }
+        				}
+	        		}
+	        		elem = metadata.get("episode");
+	        		if( elem != null ) {
+	    	        	try {
+	        				assertThat(tvEpisodeNumber).isEqualTo(String.format("%02d",elem.getAsInt()));
+	    	        	}
+	    	        	catch(NumberFormatException ex) {
```

Comment:
These catches can be chained to make the code shorter, e.g.

```
catch(NumberFormatException | NullPointerException ex) {
```

Reply:
Nifty, didn't know this idiom.